frontend/src/metabase/components/Popover.jsx

```diff
@@ -246,24 +246,25 @@ export default class Popover extends Component {
 
         this._best = best;
       }
-      if (this.props.sizeToFit) {
-        const body = tetherOptions.element.querySelector(".PopoverBody");
-        if (this._tether.attachment.top === "top") {
-          if (constrainToScreen(body, "bottom", PAGE_PADDING)) {
-            body.classList.add("scroll-y");
-            body.classList.add("scroll-show");
-          }
-        } else if (this._tether.attachment.top === "bottom") {
-          if (constrainToScreen(body, "top", PAGE_PADDING)) {
-            body.classList.add("scroll-y");
-            body.classList.add("scroll-show");
-          }
-        }
-      }
-
       // finally set the best options
       this._setTetherOptions(tetherOptions, this._best);
     }
+
+    // after tether has done its business, resize the popover body to fit the browser window if so desired
+    if (this.props.sizeToFit) {
```

Comment:
this is going to have 15502 pop back up again, it looks like, tho?

Reply:
I can't repro the shifting ot the popover but I can repro the popover covering the button after exactly 4 clicks again

Reply:
oh ya you're right, was able to repro it

Reply:
@howonlee check out the change I made here https://github.com/metabase/metabase/pull/15608/commits/6fb13700454dc06830cf8a70b5ca5081d86c3427

this hack seems to fix things for me. anything else broken for you?